### PR TITLE
Edit ministry team

### PR DESCRIPTION
**Pull request by supersun1**

### CODE REVIEW

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java [hunk 1]
```diff
@@ -31,33 +33,48 @@ protected void onCreate(Bundle savedInstanceState) {
         groupID = intent.getStringExtra("groupID");
 
         createBundle(groupID);
-        startFragment(fragmentType);
-
+        try {
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: Don't think you need a ```try {} catch() {}``` here.

File: app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java [hunk 2]
```diff
@@ -49,10 +59,36 @@ public void setupData(FormHolder formHolder)
...
+        for(CruUser user: ministryTeam.ministryTeamLeaders) {
+            if (loggedInID.equals(user.id))
+                isLeader = true;
+        }
+        return isLeader;
+    }
+
+    @OnClick(R.id.ministry_team_edit_button)
+    public void onClickUpdateMinistryTeamButton() {
+        Intent testIntent = new Intent(MinistryTeamInformationFragment.this.getContext(), UpdateGroupsInformationActivity.class);
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java]: Does ```Intent intent = new Intent(getContext(), UpdateGroupsInformationActivity.class);``` work?

Comment (tylerbwong) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java]: Also should probably rename variable ```testIntent```

File: app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java [hunk 3]
```diff
@@ -49,10 +59,36 @@ public void setupData(FormHolder formHolder)
         // retrieve the ministry team from the form holder
         ministryTeam = (MinistryTeam) formHolder.getDataObject(JoinMinistryTeamActivity.MINISTRY_TEAM);
         formHolder.setTitle(ministryTeam.name);
+//        SharedPreferencesUtil.writeLoginInformation("59bff21968325a0400be2c2d", "TestName", "TestKey");
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java]: Is this still needed?

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java (hunk 1 above)
Comment (tylerbwong) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: Don't forget to remove ```Timber``` when removing this.

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java [hunk 4]
```diff
@@ -31,33 +33,48 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        }
         System.out.println("groupID: " + groupID);
         System.out.println("fragmentType: " + fragmentType);
-
-
     }
 
     private void startFragment(String fragmentType) {
-        switch(fragmentType) {
+        switch (fragmentType) {
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: What happens in the ```default``` case?

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java [hunk 5]
```diff
@@ -31,33 +33,48 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            System.out.println("error");
             return null;
         }
         fragment.setArguments(bundle);
         return fragment;
     }
+
+    private UpdateMinistryTeamFragment createMinistryTeamFragment() {
+        UpdateMinistryTeamFragment fragment = new UpdateMinistryTeamFragment();
+        if (bundle.isEmpty()) {
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: Don't need this check.

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java [hunk 6]
```diff
@@ -31,33 +33,48 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-    private void createBundle (String groupID){
+    private void createBundle(String groupID) {
         bundle = new Bundle();
         bundle.putString("groupID", groupID);
     }
 
     private UpdateCommunityGroupFragment createCommunityGroupFragment() {
         UpdateCommunityGroupFragment fragment = new UpdateCommunityGroupFragment();
         if (bundle.isEmpty()) {
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: Don't need this check.

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateMinistryTeamFragment.java [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    public UpdateMinistryTeamFragment newInstance() {
+        return new UpdateMinistryTeamFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        compSub = new CompositeSubscription();
+        setupMinistryTeamObserver();
+        if (!getArguments().isEmpty())
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateMinistryTeamFragment.java]: What will happen in ```getArgument.isEmpty == true```?

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateMinistryTeamFragment.java [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+        teamLeaders = ministryTeam.ministryTeamLeaders;
+        teamParentMinistryID = ministryTeam.parentMinistryId;
+
+        updateTeamNameField.setText(teamName);
+        updateTeamDescriptionField.setText(teamDescription);
+
+        Picasso.with(this.getContext()).load(teamImageLink).fit().centerCrop().into(teamImageField);
+
+        teamImageField.isClickable();
+        teamImageField.setOnClickListener(new View.OnClickListener() {
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateMinistryTeamFragment.java]: ```teamImageField.setOnClickListener(view -> getNewImage());```

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateMinistryTeamFragment.java [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+        builder.setTitle("Choose Image Source");
+        builder.setItems(new CharSequence[]{"Gallery", "Camera"},
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int selectedOption) {
+                        switch (selectedOption) {
+                            case 0:
+                                getImageFromGallery();
+                                break;
+                       case 1:
```
Comment (tylerbwong) [inline app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateMinistryTeamFragment.java]: Alignment here seems off.

File: app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java (hunk 3 above)
Comment (supersun1) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java]: yes i need it to prove that the button is populated dynamically

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java (hunk 4 above)
Comment (supersun1) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: there's no default case, only two cases and is specifically for community group and ministry team

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java (hunk 6 above)
Comment (supersun1) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: im not sure what you mean not needing this check

File: app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java (hunk 3 above)
Comment (tylerbwong) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/ministryteams/MinistryTeamInformationFragment.java]: Can't you use the debugger instead?

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java (hunk 4 above)
Comment (tylerbwong) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: I think every switch statement needs a default case. Even if it just logs an error.

File: app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java (hunk 6 above)
Comment (tylerbwong) [reply app/src/main/java/com/crucentralcoast/app/presentation/views/updategroupsinformation/UpdateGroupsInformationActivity.java]: Why do you need to check if the bundle is empty. I would still try to create the fragment.

**Pull request merged**